### PR TITLE
adopt first implementor's draft 

### PR DESCRIPTION
**Pull request by moratori**
## Purpose
  - Modifications were made to comply with the [implementer draft](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0-ID1.html). The main changes are as shown in the figure below.

ID1

## Main changes

- To validate the data structure, a JSON schema was created and Ajv was installed for validation.
- The metadata structure has been updated to comply with the latest specifications.
- When responding with metadata, Accept-Language is now interpreted as an option.

### CODE REVIEW

File: src/oid4vci/types/validator.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  const schemas = ["CredentialRequestVcSdJwt", "CredentialRequestJwtVcJson"];
+
+  return validateWithSchemas<CredentialRequest>(
+    rawData,
+    schemas,
+    schemaValidator,
+  );
+};
+
+export const credentialRequestJwtVcJsonValidator = (
```
Comment (ryosuke-wakaba) [inline src/oid4vci/types/validator.ts:81]: Is this function being used?

File: src/oid4vci/types/validator.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  rawData: unknown,
+): CredentialRequestJwtVcJson => {
+  return validateWithSchemas<CredentialRequestJwtVcJson>(
+    rawData,
+    ["CredentialRequestJwtVcJson"],
+    schemaValidator,
+  );
+};
+
+export const credentialRequestVcSdJwtValidator = (
```
Comment (ryosuke-wakaba) [inline src/oid4vci/types/validator.ts:91]: Is this function being used?

**Review by ryosuke-wakaba — COMMENTED** [review]
There are some simple things to check, so please check them.

File: src/oid4vci/types/validator.ts (hunk 1 above)
Comment (moratori) [reply src/oid4vci/types/validator.ts:81]: Yes, it is used in the process here: https://github.com/moratori/OWND-Project-VCI/blob/daf378c388fc0cd6c98df3093b4f06a452179064/src/oid4vci/credentialEndpoint/CredentialIssuer.ts#L224

File: src/oid4vci/types/validator.ts (hunk 2 above)
Comment (moratori) [reply src/oid4vci/types/validator.ts:91]: https://github.com/moratori/OWND-Project-VCI/blob/daf378c388fc0cd6c98df3093b4f06a452179064/src/oid4vci/credentialEndpoint/CredentialIssuer.ts#L233

**Comment by moratori** [comment]
Upon reconsideration, I feel that the timing of the validation might be inappropriate.
I'll think about whether it's possible to perform the validation at an earlier stage.

**Comment by moratori** [comment]
In conclusion, I would like to keep the current implementation. 
During the `_issue` process, the processing is performed according to the credential format, making it convenient to perform format-specific validation at that time.

File: src/oid4vci/types/validator.ts (hunk 1 above)
Comment (ryosuke-wakaba) [reply src/oid4vci/types/validator.ts:81]: I would like to change the fact that the same validation logic is executed twice, but I hope to be able to improve this in the future.

File: src/oid4vci/types/validator.ts (hunk 2 above)
Comment (ryosuke-wakaba) [reply src/oid4vci/types/validator.ts:91]: The same is true here.

**Review by ryosuke-wakaba — APPROVED** [review]
LGTM

**Pull request merged**